src/win/fs.c

```diff
@@ -1186,15 +1186,28 @@ static void fs__unlink_rmdir(uv_fs_t* req, BOOL isrmdir) {
       if (info.dwFileAttributes & FILE_ATTRIBUTE_READONLY) {
         /* Remove read-only attribute */
         FILE_BASIC_INFORMATION basic = { 0 };
+        HANDLE write_attributes_handle;
 
         basic.FileAttributes = (info.dwFileAttributes & ~FILE_ATTRIBUTE_READONLY) |
                               FILE_ATTRIBUTE_ARCHIVE;
 
-        status = pNtSetInformationFile(handle,
+        write_attributes_handle = ReOpenFile(handle, FILE_WRITE_ATTRIBUTES,
+                                             FILE_SHARE_READ | FILE_SHARE_WRITE |
+                                             FILE_SHARE_DELETE,
+                                             FILE_FLAG_OPEN_REPARSE_POINT |
+                                             FILE_FLAG_BACKUP_SEMANTICS);
```

Comment:
Can you add a comment here explaining this works around a wine bug and link to the bug report?

As a rule, changes should not slow things down for everyone in order to work around niche bugs in niche environments, but as this is the unhappy (and therefore hopefully infrequent) path I guess we can let that slide.

Reply:
Will do in the morning. I also wonder whether FILE_DISPOSITION_IGNORE_READONLY_ATTRIBUTE may work independently of FILE_DISPOSITION_POSIX_SEMANTICS, in which case we could avoid this dance for the case where the file system does not support posix semantics. That would leave only pre-1809 windows versions in the fallback path, which are already old and will keeping getting older.